[682] RBs can update the school contact

asmega opened the pull request:
### Context

- https://trello.com/c/afX5ZJbs/682-no-ability-to-edit-school-contacts-rb-user
- Allow RBs to update the school contact

### Changes proposed in this pull request

- Adds change link to edit contact on school show page
- If headteacher contact present radios will be visible, otherwise no radios will be visible
- It will preselect headteacher contact or someone else radio depending on the current contact on the preorder info record
- It will populate the someone else fields if an existing contact for the school exists
- Changing the radio will point the preorder info record to the correct contact

#### Add change link
![image](https://user-images.githubusercontent.com/92580/93486365-05255080-f8fc-11ea-9a6f-7c0cf48a9a83.png)

#### Edit contact page
![image](https://user-images.githubusercontent.com/92580/93486662-5df4e900-f8fc-11ea-9c50-85aa398575a4.png)

### Guidance to review

There are quite a few scenarios to play with

- A school with and without a headteacher contact
- A school with and without a preorder info record
- A school with and without a contact record set on the preorder record
- The preorder record set as the headteacher contact or another contact
- Updating the record between headteacher and other contact
- Updating the record on the second contact
- Updating with the someone else contact with the headteacher email address should update headteacher contact

tonyheadford reviewed (commented):
Looks good other than when you change the headteacher contact (using their email in the other part  to update details) you then no longer get the option to select headteacher or have a different contact you can only edit the single contact - is that the intention?

asmega commented:
> Looks good other than when you change the headteacher contact (using their email in the other part to update details) you then no longer get the option to select headteacher or have a different contact you can only edit the single contact - is that the intention?

@tonyheadford good spot. I've fixed the issue and added an extra test

tonyheadford reviewed (approved):
Yep works great now
